ext/page_log/palm/palm.c

```diff
@@ -916,6 +916,21 @@ palm_handle_put(WT_PAGE_LOG_HANDLE *plh, WT_SESSION *session, uint64_t page_id,
         }                                                                                          \
     }
 
+static int
+palm_handle_get_page_ids(WT_PAGE_LOG_HANDLE *plh, WT_SESSION *session, uint64_t checkpoint_id, WT_ITEM *item){
+    PALM_KV_CONTEXT context;
+    PALM_HANDLE *palm_handle = (PALM_HANDLE *)plh;
+    PALM *palm = palm_handle->palm;
+    palm_delay(palm, session);
+    int ret;
+
+    palm_init_context(palm, &context);
+
+    ret = palm_kv_get_page_ids(&context, item, checkpoint_id);
```

Comment:
This needs to be wrapped in an lmdb transaction `    PALM_KV_RET(palm, session, palm_kv_begin_transaction(&context, palm->kv_env, false));`